Tweak command line scripts

willthefrog opened the pull request:
- enable fine grained control of command flags parsing
- switch to attribute style of accessing config options
- break down visualization function
- add support for upper case image extensions

willthefrog committed:
Break down visualization function

decouple it from IO operations

qingqing01 commented on PaddleCV/object_detection/tools/infer.py:
```diff
@@ -36,6 +37,20 @@
 logging.basicConfig(level=logging.INFO, format=FORMAT)
 logger = logging.getLogger(__name__)
 
+# FIXME should this be configurable?
```
better configurable or use save_dir of config as default.

willthefrog replied:
`save_dir` is for checkpoints, it would be confusing to save visualizations there
 also there may be hundreds of visualization files, which would clutter `save_dir`

willthefrog replied:
I'll just leave it like this for now. please add a command line flag to your PR now that the flag parser is more flexible. @heavengate

willthefrog replied:
this is now a command line flag

willthefrog committed:
Tweak command line scripts

enable fine grained control of command flags parsing
switch to attribute style of accessing config options

heavengate reviewed (commented):
LGTM

heavengate commented on PaddleCV/object_detection/tools/infer.py:
```diff
@@ -53,37 +65,52 @@ def get_test_images(infer_dir, infer_img):
+        "--infer_img",
+        type=str,
+        default=None,
+        help="Image path, has higher priority over --image-dir")
```
这里--image-dir 麻烦帮忙成--infer_dir

willthefrog replied:
done

qingqing01 commented on PaddleCV/object_detection/tools/infer.py:
```diff
@@ -53,37 +65,52 @@ def get_test_images(infer_dir, infer_img):
+        type=str,
+        default="output",
+        help="Directory for saving visualization files.")
+    args = parser.parse_args()
```
better to put these `parser.add_argument` in another function or move to `if __name__ == '__main__'` to make the `main()` simple and clean.

willthefrog replied:
done

qingqing01 commented on PaddleCV/object_detection/tools/train.py:
```diff
@@ -40,33 +40,53 @@
+        "--savefile",
+        default=None,
+        type=str,
+        help="Evaluation file name, default to bbox.json and mask.json."
```
This arg is not needed in train and eval? only need in infer.

willthefrog replied:
it is used by `eval_results`

willthefrog committed:
Move flag parsing out of `main()`

willthefrog committed:
Fix a bug where `None` is returned instead of `{}`

willthefrog committed:
Rename `save_xxx` to `output_xxx` in command line flags

could be confusing since checkpoint is stored in `save_dir`

willthefrog committed:
Support image file extensions in upper case
